### DISCUSSION TITLE
Unable to migrate Xamarin.iOS binding project

### DISCUSSION THREAD

**Discussion by nk-alex** [dotnet/maui#23157] [Upgrading from Xamarin]
I'm following this [article](https://learn.microsoft.com/en-us/dotnet/maui/migration/ios-binding-projects?view=net-maui-8.0) in order to migrate a Xamarin.iOS binding library to MAUI.

Back in Xamarin.Forms, by creating a iOS Bindings Library (Xamarin), a project like this is generated: 

![image](https://github.com/dotnet/maui/assets/101177816/80ce64e3-772d-4711-947b-95ebf2b0de90)

Native references could be added by right click on native references.

In MAUI, when I create a new iOS Binding Library, this is the project generated:

![image](https://github.com/dotnet/maui/assets/101177816/4b50f5ac-6a94-4dee-9da9-6f7090c992dd)

The step 2 in the mentioned [article](https://learn.microsoft.com/en-us/dotnet/maui/migration/ios-binding-projects?view=net-maui-8.0), says `Add your native library, or native framework, to the project as a native reference` But I'm not able to do it. If I try to add my .a native library as project reference I receive `The reference is invalid or unsupported.`
If I edit the project myself adding:

```
<ItemGroup>
    <NativeReference Include="library.a">
        <Kind>Static</Kind>
        <ForceLoad>True</ForceLoad>
    </NativeReference>
</ItemGroup>
```

the project compiles but I cannot use the generated .dll file.

**Comment by nk-alex** [comment]
https://github.com/dotnet/maui/issues/14775

**Reply by munkii** [reply]
#14775 does not seem like the same issue to me.

I too am trying to migrate an exisitng iSO XF binding library to a MAUI project. I am aware of the existing issue (from at least 2021) that means you cannot reference the Project directly you have to use the dll

The problem I am having is that VS 2022 IDE just seems to ignore any attempt to add a Native reference. 

In [Xamarin.iOS binding project migration](https://learn.microsoft.com/en-us/dotnet/maui/migration/ios-binding-projects?view=net-maui-8.0) Step 2 says "Add your native library, or native framework, to the project as a native reference". How do I do that? There is no option to add a native framework in the IDE and any attempt to "Add New Item" just expects me to add a source file.

I see nothing like this menu that was in XF

![image](https://github.com/dotnet/maui/assets/827342/e610e5b1-d2d4-440f-89be-d35fc621c903)

Step 2 then says "update the project file from your Xamarin.iOS binding project" I assume that means simply copy the text element <NativeReference> and into your new CSPROJ file. I've done that and now I have 

```
	<ItemGroup>
		<NativeReference Include="Native Reference/TheProduct.framework">
			<Kind>Framework</Kind>
		</NativeReference>
	</ItemGroup>

```

However that is not reflected in the IDE I just see

![image](https://github.com/dotnet/maui/assets/827342/08323a57-c791-4ce3-9c09-4f7fe1b9387b)

The new binding library builds but looking at its ouput dll in Object Browser there are no types

Here is the complete CSPRoj

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net8.0-ios</TargetFramework>
    <Nullable>enable</Nullable>
    <ImplicitUsings>true</ImplicitUsings>
    <IsBindingProject>true</IsBindingProject>
  </PropertyGroup>

  <ItemGroup>
    <ObjcBindingApiDefinition Include="ApiDefinition.cs" />
    <ObjcBindingCoreSource Include="StructsAndEnums.cs" />
  </ItemGroup>
	<ItemGroup>
		<NativeReference Include="Native Reference/TheProduct.framework">
			<Kind>Framework</Kind>
		</NativeReference>
	</ItemGroup>
</Project>

```
